proxy/proxy.go

```diff
@@ -847,6 +864,52 @@ func WithParams(p Params) *Proxy {
 			maxUnhealthyEndpointsRatio: p.PassiveHealthCheck.MaxUnhealthyEndpointsRatio,
 		}
 	}
+
+	log := &logging.DefaultLog{}
+
+	var (
+		frURL *url.URL
+		// buffered channel size 10k to allow unblocked requests
+		frChannel = make(chan struct{}, 10240)
+	)
+	if p.FlightRecorder != nil {
+		var err error
+		frURL, err = url.Parse(p.FlightRecorderTargetURL)
+		if err != nil {
+			p.FlightRecorder.Stop()
+			p.FlightRecorder = nil
+		} else {
+			// decouple writing a trace from data-plane work
+			go func() {
+				weekly := 7 * 24 * time.Hour
+				timer := time.NewTimer(weekly)
```

Comment:
Why do we need this timer?
I suspect this is some kind of optimisation that I don't understand yet.

Reply:
But this will hang forever  as well, i guess?
```go
				for {
					select {
					case <-frChannel:
						if time.Since(last) >= time.Hour {
							writeGoTrace(p.FlightRecorder, frURL, log, tr)
						}
						last = time.Now()
					case <-quit:
						p.FlightRecorder.Stop()
						return
					}
				}
```
The only difference is that with timer it will write trace one in a week even without slow request.

Reply:
Then we have the write not decoupled because the other side of the channel is the hot path and can fill the buffered channel because we are on a blocked write 


We can replace week with 100y and call it "forever"